### PR TITLE
Submit analysis functionality

### PR DESCRIPTION
**Pull request by vincerubinetti**
Roughly in order of diff:

- add fixture data for mocking api calls, and hook them up to msw. useful for testing more quickly and avoiding cloud func costs.
- update packages (except very recent eslint v9 breaking change)
- redirect to home route on 404
- add basic api func to call ml endpoint
- add type schema for analysis results
- add stricter typing for things like species (now union of possible options instead of just string)
- remove some components/other stuff specific to molevolvr
- fix controlled bug in some zag components by moving value and onValueChange to declarative context instead of machine initial state
- tweak radio component appearance and types
- tweak select component types. change from value being full Option to just id (easier to sync with url and handle in other places). update all components where used.
- simplify table filter functions
- add "shrinkWrap" util func and use it on tooltip
- add basic "load analysis" page. either takes inputs sent to it from "new analysis" page and runs query, or allows user to upload results directly. includes download button. has very basic temporary display of analysis results.
- complete rest of "new analysis" page
    - add select/dropdowns for network type and negatives set params
    - include node/edge counts under network buttons, dependent on selected species
    - add submit analysis button that sends state to "load analysis" page
    - restricts species based on selected network type and negatives set params, and show warnings
- add condition to testbed to test toasts for react strict-mode's double-triggering of useEffect
- util func tweaks

### CODE REVIEW

**Review by falquaddoomi — COMMENTED** [review]
Nice, glanced over the code and tried it out locally, and things are looking good. I had a comment about the 404 redirect, but aside from that I think this is good to go.

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -101,25 +100,27 @@ const routes = [
...
         path: "testbed",
         element: <Testbed />,
         loader: () => ({ toc: true }) satisfies Meta,
       },
+      {
+        path: "*",
+        loader: async () => {
+          return redirect("/");
+        },
+      },
```
[the remark this answers is not in the document]
Comment (vincerubinetti) [reply frontend/src/App.tsx]: Thanks, I didn't see any other branch or a fork you made though, did you push it? Either way I added this in the latest commit.

**Review by falquaddoomi — APPROVED** [review]
Nice, thanks for entertaining my suggestion! I like that little "broken image" bit on the 404 page.

**Review by ChristopherMancuso — APPROVED** [review]
I think this is looking great so far. I check it out and I think the buttons are looking good and it did seem to run a job and give that basic output out.  A few things to change in the future

- In section 3 Chose Parameters, change "Negatives Source" to "Geneset Context"
- For the example gene set a user can load. At some point will probably need any example gene set for each species. 
- I think having the "Species" in section 3 probably will have to change. I'm thinking that may even have to be moved to the very top and the first thing a user does is chose a training species and a prediction species. I can mock up something about what I'm thinking and I don't think it will require that big of a change.

This is def good to move onto a first pass through the results pages!

**Comment by vincerubinetti** [comment]
> * In section 3 Chose Parameters, change "Negatives Source" to "Geneset Context"

I can change this now.

> * For the example gene set a user can load. At some point will probably need any example gene set for each species.

If you want to take that as an action item, it will be easy to add later.

> * I think having the "Species" in section 3 probably will have to change. I'm thinking that may even have to be moved to the very top and the first thing a user does is chose a training species and a prediction species. I can mock up something about what I'm thinking and I don't think it will require that big of a change.

I thought for now we were going with keeping them the same always? Doing two separate species select is no problem for me though. Perhaps it makes sense to keep it how it is now -- with a species select in section 1 and a species select in section 3 -- but instead of linking them together as they are now, they can be separate state and labeled separately. I don't exactly understand what "prediction" vs "training" species means, but perhaps one of those makes sense to be next to the genes the user inputs, and the other makes sense to be next to the other param selection.

Regardless, for now, we'll keep it as is, and you can share any ideas/sketches you have for the future.

**Pull request merged**